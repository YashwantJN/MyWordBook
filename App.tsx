import React from 'react';
import AuthContextProvider from './context/Auth.context';
import {RootNavigators} from './navigation/RootNavigator';

const App = () => {
  return (
    <AuthContextProvider>
      <RootNavigators />
    </AuthContextProvider>
  );
};

export default App;
