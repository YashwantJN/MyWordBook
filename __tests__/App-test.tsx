/**
 * @format
 */

import 'react-native'
import React from 'react'
import App from '../App'
import renderer, { act } from 'react-test-renderer'

jest.mock('react-native-sound', () => {
  return {
    Sound: jest.fn(),
  }
})

test('app renders correctly', async () => {
  const tree = renderer.create(<App />).toJSON()
  await act(async () => {
    expect(tree).toMatchSnapshot()
  })
})
