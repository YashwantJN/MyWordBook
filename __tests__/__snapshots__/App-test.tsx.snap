// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Module app renders correctly for un-auth 1`] = `
Array [
  <RCTSafeAreaView
    style={
      Object {
        "backgroundColor": "#183d3d",
        "flex": 0,
      }
    }
  />,
  <RCTSafeAreaView
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "flex": 1,
      }
    }
  >
    <RNCSafeAreaProvider
      onInsetsChange={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenStack
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          collapsable={false}
          forwardedRef={[Function]}
          gestureResponseDistance={
            Object {
              "bottom": -1,
              "end": -1,
              "start": -1,
              "top": -1,
            }
          }
          nativeBackButtonDismissalEnabled={false}
          onAppear={[Function]}
          onDisappear={[Function]}
          onDismissed={[Function]}
          onGestureCancel={[Function]}
          onHeaderBackButtonClicked={[Function]}
          onNativeDismissCancelled={[Function]}
          onTransitionProgress={[Function]}
          onWillDisappear={[Function]}
          replaceAnimation="push"
          sheetAllowedDetents="large"
          sheetCornerRadius={-1}
          sheetExpandsWhenScrolledToEdge={true}
          sheetGrabberVisible={false}
          sheetLargestUndimmedDetent="all"
          stackPresentation="push"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          swipeDirection="horizontal"
        >
          <View
            accessibilityElementsHidden={false}
            importantForAccessibility="auto"
            style={
              Object {
                "flex": 1,
                "flexDirection": "column-reverse",
              }
            }
          >
            <View
              collapsable={false}
              style={
                Array [
                  Object {
                    "flex": 1,
                  },
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#FFFFFF",
                    "flex": 1,
                    "justifyContent": "center",
                    "padding": 16,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#183d3d",
                      "fontSize": 28,
                      "fontWeight": "500",
                      "letterSpacing": 1,
                      "textAlign": "center",
                    }
                  }
                >
                  Thank you for choosing
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#ffd966",
                      "fontSize": 32,
                      "fontWeight": "600",
                      "letterSpacing": 1.5,
                      "paddingTop": 5,
                      "textAlign": "center",
                    }
                  }
                >
                  MY BOOK
                </Text>
                <Image
                  source={
                    Object {
                      "testUri": "../../../src/assets/images/girlWithBook.png",
                    }
                  }
                  style={
                    Object {
                      "height": "50%",
                      "resizeMode": "contain",
                      "width": "100%",
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "color": "#183d3d",
                      "fontSize": 14,
                      "fontWeight": "200",
                      "letterSpacing": 1,
                      "textAlign": "center",
                    }
                  }
                >
                  Explore the world of words!
                </Text>
                <View
                  accessibilityState={
                    Object {
                      "disabled": false,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "backgroundColor": "#ffd966",
                      "borderColor": "#aac8a7",
                      "borderRadius": 36,
                      "borderWidth": 2,
                      "justifyContent": "center",
                      "marginTop": 16,
                      "opacity": 1,
                      "paddingHorizontal": 32,
                      "paddingVertical": 14,
                      "width": "100%",
                    }
                  }
                  testID="primaryButton"
                >
                  <Text
                    style={
                      Object {
                        "color": "#040d12",
                        "fontSize": 18,
                        "fontWeight": "600",
                        "letterSpacing": 2,
                        "textAlign": "center",
                      }
                    }
                  >
                    READ &gt;&gt;
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <RNSScreenStackHeaderConfig
            backButtonInCustomView={false}
            backTitle="Back"
            backTitleVisible={true}
            backgroundColor="#183d3d"
            color="#93b1a6"
            direction="ltr"
            disableBackButtonMenu={false}
            hidden={true}
            hideBackButton={false}
            largeTitleHideShadow={false}
            title="Welcome"
            titleColor="#93b1a6"
            topInsetEnabled={false}
            translucent={false}
          />
        </RNSScreen>
      </RNSScreenStack>
    </RNCSafeAreaProvider>
  </RCTSafeAreaView>,
]
`;
